torch/csrc/jit/passes/utils/subgraph_utils.cpp

```diff
@@ -362,6 +432,174 @@ Node* createSingletonSubgraphAndUpdateAliasing(
       });
 }
 
+bool unmergeOutputsAilisingInputs(Node* subgraphNode) {
+  GRAPH_DEBUG("unfuseOutputsAilisingInputs on ", getHeader(subgraphNode));
+  auto subgraph = subgraphNode->g(attr::Subgraph);
+  AliasDb alias_db(subgraph);
+
+  std::set<Node*, topo_cmp_node> nodes;
+  for (auto o : subgraph->outputs()) {
+    if (alias_db.mayContainAlias({o}, subgraph->inputs())) {
+      collectNodesToUnfuse(o->node(), nodes);
+    }
+  }
+
+  // unfuse in the reverse topo order
+  for (auto it = nodes.rbegin(); it != nodes.rend(); it++) {
+    SubgraphUtils::unmergeNode(*it, subgraphNode);
+  }
+
+  return !nodes.empty();
+}
+
+bool unmergeAliasedOutputs(Node* subgraphNode) {
+  GRAPH_DEBUG("unfuseAliasedOutputs on ", getHeader(subgraphNode));
+  if (subgraphNode->outputs().size() < 2) {
+    return false;
+  }
+
+  auto subgraph = subgraphNode->g(attr::Subgraph);
+  GRAPH_DUMP("unfuseAliasedOutputs Subgraph ", subgraph);
+  auto sets = buildAliasedSets(subgraph);
```

Comment:
would be great if we can only rebuild this in the case that we've merged out a list of tensors, but i guess in a follow up.

Reply:
agreed, I'll try following up ASAP on this.